solvers/qpSpline/nWaypointCubicSplinemex.cpp

```diff
@@ -8,30 +8,33 @@ using namespace std;
 using namespace Eigen;
 
 void mexFunction(int nlhs, mxArray *plhs[],int nrhs, const mxArray *prhs[]) {
-  string usage = "[coefs, objval] = twoWaypointCubicSplinemex(ts, xs, xd0, xdf)";
+  string usage = "[coefs, objval] = nWaypointCubicSplinemex(ts, xs, xd0, xdf)";
   if (nrhs != 4)
-    mexErrMsgIdAndTxt("Drake:twoWaypointCubicSplinemex:WrongNumberOfInputs", usage.c_str());
+    mexErrMsgIdAndTxt("Drake:nWaypointCubicSplinemex:WrongNumberOfInputs", usage.c_str());
   if (nlhs > 2)
-    mexErrMsgIdAndTxt("Drake:twoWaypointCubicSplinemex:WrongNumberOfOutputs", usage.c_str());
+    mexErrMsgIdAndTxt("Drake:nWaypointCubicSplinemex:WrongNumberOfOutputs", usage.c_str());
 
-  const std::vector<double> segment_times = matlabToStdVector(prhs[0]);
+  const std::vector<double> segment_times = matlabToStdVector<double>(prhs[0]);
   MatrixXd xs = matlabToEigen<Dynamic, Dynamic>(prhs[1]);
   auto xd0 = matlabToEigen<Dynamic, 1>(prhs[2]);
   auto xdf = matlabToEigen<Dynamic, 1>(prhs[3]);
 
   mwSize ndof = static_cast<mwSize>(xs.rows());
-  mwSize num_segments = 3;
+  mwSize num_segments = static_cast<mwSize>(xs.cols())-1;
+  mwSize num_knots = num_segments - 1;
   mwSize num_coeffs_per_segment = 4;
   mwSize dims[] = {ndof, num_segments, num_coeffs_per_segment};
-  plhs[0] = mxCreateNumericArray(num_segments, dims, mxDOUBLE_CLASS, mxREAL);
+  plhs[0] = mxCreateNumericArray(3, dims, mxDOUBLE_CLASS, mxREAL);
```

Comment:
Yeah, it was a bug before, but it worked because num_segments was always 3. This 3 is the number of dimensions of the result, so it's definitely independent of num_segments